fdbserver/MoveKeys.actor.cpp

```diff
@@ -907,9 +911,12 @@ ACTOR Future<Void> checkFetchingState(Database cx,
 			std::vector<Future<Void>> requests;
 			state std::vector<Future<Void>> tssRequests;
 			for (int s = 0; s < serverListValues.size(); s++) {
+				// We don't think this condition should be triggered, but we're not sure if there are conditions
+				// that might cause it to trigger. Adding this assertion to find any such cases via testing.
+				ASSERT_WE_THINK(serverListValues[s].present());
```

Comment:
@sfc-gh-akejriwal, I saw this assertion triggered in a test:

```
 ./fdbserver-7.1.27 -r simulation -f ./tests/restarting/from_7.1.0_until_7.2.0/ConfigureTestRestart-1.toml -s 1969136453 -b on --logsize 1GB
./fdbserver -r simulation -f ./tests/restarting/from_7.1.0_until_7.2.0/ConfigureTestRestart-2.toml --restarting -s 1969136454 -b on
```
The `fdbserver` is at commit c423a60e5 with clang.

Reply:
The sequence events are:

SS 9dcb61e46de25b88 Machine="2.0.1.0:1"
- Shard move started at 469.239433
```
469.239433 SendRelocateToDDQueue ID=e93b602096d1e0fe SuppressedEventCount=0 TraceId=b5b0a1d153452ecd ServerPrimary=1 ServerTeam=TeamID 31f422f6663842b1; Size 1; 2.0.1.2:1 80d17161588d3566 KeyBegin= KeyEnd=\xff\xff Priority=700 ServerTeamFailedMachines=0 ServerTeamOKMachines=1 BK,CP,CS,DD,GP,MS,RK,SS,TL
469.239533 RelocateShard ID=e93b602096d1e0fe BeginPair=b5b0a1d153452ecd KeyBegin= KeyEnd=\xff\xff Priority=700 SuppressedEventCount=0 BK,CP,CS,DD,GP,MS,RK,SS,TL
469.239533 RelocateShardHasDestination ID=e93b602096d1e0fe PairId=b5b0a1d153452ecd Priority=700 KeyBegin= KeyEnd=\xff\xff SourceServers=730258fd3da1eb6b,80d17161588d3566 DestinationTeam=9dcb61e46de25b88,730258fd3da1eb6b ExtraIds=[no items] BK,CP,CS,DD,GP,MS,RK,SS,TL
469.239533 RelocateShard_StartMoveKeys ID=b5b0a1d153452ecd BeginPair=7aa096b9cbd44cd0 BK,CP,CS,DD,GP,MS,RK,SS,TL
```

- SS removed at 469.245336
```
469.245336 StorageServerWorkerRemoved ID=9dcb61e46de25b88 Reason=ServerTag MutationType=ClearRange TagMatches=1 IsTSS=0 BK,CC,CP,RV,SS,TL
469.245336 KVClose ID=9dcb61e46de25b88 Filename=/root/simfdb/b10b2a8f804057114a229bfef714d3ea/storage-9dcb61e46de25b8853e9a9c6b517328d.sqlite Del=1 BK,CC,CP,RV,SS,TL
469.245336 WorkPool_Stop ID=0000000000000000 Error=success ErrorDescription=Success ErrorCode=0 Workers=64 Idle=64 Work=0 BK,CC,CP,RV,SS,TL
469.245336 WorkPool_Stop ID=0000000000000000 Error=success ErrorDescription=Success ErrorCode=0 Workers=1 Idle=1 Work=0 BK,CC,CP,RV,SS,TL
469.245336 StorageServerTerminated ID=9dcb61e46de25b88 Error=worker_removed ErrorDescription=Normal worker shut down ErrorCode=1202 BK,CC,CP,RV,SS,TL
```

- DD noticed SS removal at 469.257334
```
469.257334 StatusMapChange ID=e93b602096d1e0fe ServerID=9dcb61e46de25b88 Status=Removed BK,CP,CS,DD,GP,MS,RK,SS,TL
469.257334 RemovedStorageServer ID=e93b602096d1e0fe ServerID=9dcb61e46de25b88 BK,CP,CS,DD,GP,MS,RK,SS,TL
469.257334 ServerTeamRemoved ID=0000000000000000 Primary=1 TeamServerIDs=9dcb61e46de25b8853e9a9c6b517328d  TeamID=8117fa285c3980b1 BK,CP,CS,DD,GP,MS,RK,SS,TL
469.257334 RemovedServerTeam ID=e93b602096d1e0fe Team=TeamID 8117fa285c3980b1; Size 1; 2.0.1.0:1 9dcb61e46de25b88 BK,CP,CS,DD,GP,MS,RK,SS,TL
469.257334 TeamTrackerStopping ID=e93b602096d1e0fe Error=operation_cancelled ErrorDescription=Asynchronous operation cancelled ErrorCode=1101 ServerPrimary=1 Team=TeamID 8117fa285c3980b1; Size 1; 2.0.1.0:1 9dcb61e46de25b88 Priority=700 BK,CP,CS,DD,GP,MS,RK,SS,TL
```

- Assertion triggered at 469.272647

Reply:
So I think this assertion should be reverted, since this can actually happen.

Reply:
Thanks for finding this! I'm looking into it this afternoon and will revert the assertion.

Reply:
The assertion was triggered at line 712, so the above comments were misplaced.